src/ray/gcs/gcs_client/usage_stats_client.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/gcs/gcs_client/usage_stats_client.h"
+
+namespace ray {
+namespace gcs {
+UsageStatsClient::UsageStatsClient(const std::string &gcs_address,
+                                   instrumented_io_context &io_service)
+    : io_service_(io_service) {
+  GcsClientOptions options(gcs_address);
+  gcs_client_ = std::make_unique<GcsClient>(options);
+}
+
+void UsageStatsClient::RecordExtraUsageTag(const std::string &key,
+                                           const std::string &value) {
+  if (!gcs_client_->IsConnected()) {
+    RAY_CHECK_OK(gcs_client_->Connect(io_service_));
+  }
+
+  RAY_CHECK_OK(gcs_client_->InternalKV().AsyncInternalKVPut(
+      kUsageStatsNamespace,
+      kExtraUsageTagPrefix + key,
+      value,
+      /*overwrite=*/true,
+      [](Status status, boost::optional<int> added_num) {
+        if (!status.ok()) {
+          RAY_LOG(DEBUG) << "Failed to put extra usage tag, status = " << status;
```

Comment:
shouldn't this be  RAY_LOG(ERROR) ?

Reply:
For usage stats we log debug given this is a harmless error: usage stats doesn't affect the normal use of Ray.

Reply:
Hmmm, I think it's still an error and it shouldn't happen a lot (otherwise GCS is down I think?)
Do you think it's OK to change it to `RAY_LOG_EVERY_N(ERROR, 100)` maybe?